packages/yuudachi/src/functions/logging/forwardReport.ts

```diff
@@ -0,0 +1,45 @@
+import type { Message, User } from "discord.js";
+import { inlineCode, userMention } from "discord.js";
+import i18next from "i18next";
+import type { Report } from "../reports/createReport.js";
+import { updateReport } from "../reports/updateReport.js";
+import { checkReportForum } from "../settings/checkLogChannel.js";
+import { getGuildSetting, SettingsKeys } from "../settings/getGuildSetting.js";
+import { formatMessageToEmbed } from "./formatMessageToEmbed.js";
+
+type ForwardReportData = {
+	author: User;
+	reason: string;
+};
+
+export async function forwardReport(
+	{ author, reason }: ForwardReportData,
+	message: Message<true>,
+	report: Report,
+	locale: string,
+): Promise<void> {
+	const channel = checkReportForum(
+		message.guild,
+		await getGuildSetting(message.guild.id, SettingsKeys.ReportChannelId),
+	);
+
+	const thread = await channel!.threads.fetch(report.logPostId!);
+
+	await updateReport({
+		reportId: report.reportId,
+		guildId: report.guildId,
+		contextMessagesIds: [...report.contextMessagesIds, message.id],
+	});
+
+	await thread!.send({
```

Comment:
There seems to be no fallback here in case the thread doesn't exist or there is an error fetching it.

Reply:
Fixed in b6bf6a2